src/config.rs

```diff
@@ -0,0 +1,181 @@
+//! Implementation of the configuration module.
+
+use anyhow::Context;
+use anyhow::Result;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::Mode;
+use crate::commands::Commands;
+use crate::commands::check;
+use crate::commands::format;
+use crate::commands::validate;
+/// Represents the configuration for the Sprocket CLI tool.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct Config {
+    /// Configuration for the `format` command.
+    #[serde(rename = "format")]
+    pub format_config: FormatConfig,
```

Comment:
Nit: I think `_config` is a bit redundant given the type name, so perhaps just `format`, `check`, etc. for field names?